lib/elixir/test/elixir/enum_test.exs

```diff
@@ -2418,35 +2408,37 @@ defmodule EnumTest.Map do
     # set a fixed seed so the test can be deterministic
     # please note the order of following assertions is important
     map = %{a: 1, b: 2, c: 3}
+    [x1, x2, x3] = Map.to_list(map)
     seed1 = {1406, 407_414, 139_258}
     seed2 = {1406, 421_106, 567_597}
     :rand.seed(:exsss, seed1)
-    assert Enum.take_random(map, 1) == [c: 3]
+    assert Enum.take_random(map, 1) == [x3]
```

Comment:
I took this approache here, but I'm not sure if it's the best here. Maybe these tests could actually be removed and more property-based testing could be added in StreamData?

Reply:
These tests are silly in a way but important because we often have special clauses for lists, so we need to test the enum implementations. The other option is to use `MapSet`, but they don't guarantee order as well, so...

If we had a queue :D

Reply:
Oh, I see!
If we want something that's a) ordered and b) not optimized (rules out ranges), I think streams could be a nice candidate? Something like:
https://github.com/elixir-lang/elixir/compare/main...sabiwara:elixir:use-stream-for-enum-tests?expand=1

If there's interest, I can open up a PR.

Noted for the queue ;) (but this might be optimizeable too)

Reply:
Yes, streams are an option and we have tests with them too. But I would only add tests, the changes you did are great.

Reply:
So basically keep `EnumTest.Map` but add `EnumTest.Stream` as well?

Reply:
If necessary, surely, but i assume we are well covered? :)